src/qml/EmbeddedFeatureForm.qml

```diff
@@ -0,0 +1,61 @@
+import QtQuick 2.11
+import QtQuick.Controls 2.4
+
+import QtQuick.Layouts 1.0
+import "."
+import Theme 1.0
+
+import org.qfield 1.0
+import org.qgis 1.0
+
+Popup {
+    id: formPopup
+
+    //the add entry stuff
+    property var attributeFormModel: AttributeFormModel {
+        featureModel: FeatureModel {
+        }
+    }
+
+    property var state
+
+    signal featureSaved
+    signal featureCancelled
+
+    onStateChanged: {
+        //because the binding in form does not work
```

Comment:
@3nids 
That's something I don't like. I didn't invest more time on it, so you can continue your work.
Tell me if you have an idea why the binding does not work here...

Reply:
@3nids Regarding your question to bind it directly. 
```
embeddedPopup.form.state = !readOnly ? 'Edit' : 'ReadOnly'
```
This leads to an issue, since it cannot connect to the `form`.
It would be possible to connect to it, when having it like this:
```
property FeatureForm form : FeatureForm {
```
But it leaded to issues with the data. An empty form appeared :thinking:

Reply:
```embeddedPopup.form.state = !readOnly ? 'Edit' : 'ReadOnly'```

You cannot access properties of embedded objects.
You can however "proxy" them, by making a toplevel `property bool readonly isReadOnly: form.state == readOnly` parameter which can be accessed from outside (readonly is optional).

Reply:
Why not the `embeddedPopup.state` and bind it internally to the `embeddedPopup.form.state` ?

Reply:
That should also work.
I also remembered the right tool to use again: `property alias`. https://doc.qt.io/qt-5/qtqml-syntax-objectattributes.html

Reply:
This works:
```
id: formPopup
property alias superstate: form.state
FeatureForm {
        id: form
}
```
This does not work:
```
id: formPopup
property var superstate

FeatureForm {
        id: form
	state: formPopup.superstate
```
Suddenly the binding is broken. Why?

Somewhere in the FeatureForm it says:
```
state = 'Edit'
```

So the conclusion: Since an alias binds on both sides a binding does not break when we set the "depending" object (like state = 'Edit'). @m-kuhn, does that makes sence?

Reply:
makes sense to me

Reply:
Thanks. It's kind of the illumination I was looking for. I guess I will use more often aliases in future...

Reply:
Yes. This is called "breaking the binding" and there is a tool since Qt 5.10 to detect it

https://www.kdab.com/new-qt-5-10-diagnostics-breaking-qml-bindings/

It would be really good if we checked this at some point.
> # Are broken bindings a problem?
> 
> 
> 
> In general, the above example is 100% correct QML code, following the documented QML semantics. Therefore, one cannot claim that such code should be rejected by the QML engine or unconditionally raise warnings.
> 
> However, in my experience at KDAB, *I have never encountered one single occasion when breaking a binding was intended by the developer*. Broken bindings were always caused by accident (during refactorings, not realizing that the original property was bound to something, etc.), or by excessive usage of JavaScript to manage state, instead of using more declarative approaches.

Reply:
>  I have never encountered one single occasion when breaking a binding was intended by the developer.

Yes. But in this case we need to set the `state` from different places and we need the proxy property to pass it to the child object. Means the `alias` Property is exactly what we need.

Would be nice to have it in Qt Creator to show when a binding is broken. Because as you say, it's usually not intended.

Reply:
The quote is from the link above, not even emphasis is mine.